### PR TITLE
add solution for day4

### PR DESCRIPTION
**Pull request by shraddhaag**
This PR adds solution to AoC 2023 day 4.

To run the solution, navigate to the `day4` directory, and run

```
go run main.go <PATH_TO_YOUR_INPUT_FILE>
```

Today was fairly easy as compared to previous days!

### CODE REVIEW

File: day4/main.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+				winPoints *= 2
+			}
+		}
+	}
+
+	return winPoints
+}
+
+func isDigit(char byte) bool {
+	if char >= 48 && char <= 57 {
```
Comment (teivah) [inline day4/main.go]:
```suggestion
	if char >= '0' && char <= '9' {
```

Comment (teivah) [reply day4/main.go]: Just a minor readability comment

File: day4/main.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+
+	return output
+}
+
+func findNumbers(input string) int {
```
Comment (alexjoedt) [inline day4/main.go]: you could return here the result and the matches to use this in findNumberOfCards
`func findNumbers(input string) (int, int)`

**Review by alexjoedt — CHANGES_REQUESTED** [review]
```go
func findNumberOfCards(input []string) int {
	index := 0
	for {
		if index >= len(input) {
			break
		}

		parts := strings.Fields(input[index])
		cardId, err := strconv.Atoi(strings.TrimSuffix(parts[1], ":"))
		if err != nil {
			panic(err)
		}
		matches := findNumberOfWinNumbers(input[index])
		for i := cardId; i < (cardId + matches); i++ {
			input = append(input, input[i])
		}

		index++
	}

	return index
}
```

File: day4/main.go (hunk 1 above)
Comment (shraddhaag) [reply day4/main.go]: Replaced this with `unicode.IsDigit`.

Comment (teivah) [reply day4/main.go]: Even better :)

**Pull request merged**